pkg/base1/cockpit.js

```diff
@@ -1034,6 +1034,31 @@ function basic_scope(cockpit, jquery) {
         uri: calculate_url,
     };
 
+    /* ------------------------------------------------------------------------------------
+     * An ordered queue of functions that should be called later.
+     */
+
+    var later_queue = [];
+    var later_timeout = null;
+
+    function later_drain() {
+        var func, queue = later_queue;
+        later_timeout = null;
+        later_queue = [];
+        for (;;) {
+            func = queue.shift();
+            if (!func)
+                break;
```

Comment:
This isn't wrong per se, since as you stated the callback functions here shouldn't ever be null, but wouldn't `continue` be more appropriate?